pkg/api/controllers/k3smetadata/settings_handler.go

```diff
@@ -0,0 +1,69 @@
+package k3smetadata
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/rancher/rancher/pkg/channelserver"
+	"github.com/rancher/types/apis/management.cattle.io/v3"
+	"github.com/rancher/types/config"
+	"github.com/rancher/wrangler/pkg/data"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+const (
+	hashAnnotation = "field.cattle.io/contentHash"
+)
+
+var (
+	url, interval = channelserver.GetURLAndInterval()
+	hash string
+	rancherCtx context.Context
+)
+
+type handler struct {
+	ctx context.Context
+}
+
+func Register(ctx context.Context, scaled *config.ScaledContext) {
+	h := &handler{
+		ctx: ctx,
+	}
+	scaled.Management.Settings("").AddHandler(ctx, "channelserver-restart-handler", h.sync)
+	rancherCtx = ctx
+}
+
+func (h *handler) sync(key string, setting *v3.Setting) (runtime.Object, error) {
+	if setting == nil || setting.DeletionTimestamp != nil {
```

Comment:
you should also short circuit if the setting.name is not the one youre lookin for